node/consensus/utils.go

```diff
@@ -29,3 +36,78 @@ func printEvent(event []byte) string {
 	}
 	return ce.String()
 }
+
+func CreateDataCommonBlock(blockNum uint64, prevHash []byte, batchID arma_types.BatchID, decisionNum arma_types.DecisionNum, batchCount, batchIndex int, lastConfigBlockNum uint64) (*common.Block, error) {
+	block := protoutil.NewBlock(blockNum, prevHash)
+	block.Header.DataHash = batchID.Digest()
+	blockMetadata, err := ledger.AssemblerBlockMetadataToBytes(batchID, &state.OrderingInformation{DecisionNum: decisionNum, BatchCount: batchCount, BatchIndex: batchIndex}, 0)
+	if err != nil {
+		return nil, errors.Errorf("Failed to invoke AssemblerBlockMetadataToBytes: %s", err)
+	}
+	block.Metadata.Metadata[common.BlockMetadataIndex_ORDERER] = blockMetadata
+	block.Metadata.Metadata[common.BlockMetadataIndex_LAST_CONFIG] = protoutil.MarshalOrPanic(&common.Metadata{
+		Value: protoutil.MarshalOrPanic(&common.LastConfig{Index: lastConfigBlockNum}),
+	})
+	return block, err
+}
+
+func CreateConfigCommonBlock(blockNum uint64, prevHash []byte, decisionNum arma_types.DecisionNum, batchCount, batchIndex int, configReq []byte) (*common.Block, error) {
+	configBlock := protoutil.NewBlock(blockNum, prevHash)
+	configBlock.Data = &common.BlockData{Data: [][]byte{configReq}}
+	configBlock.Header.DataHash = protoutil.ComputeBlockDataHash(configBlock.Data)
```

Comment:
We should switch to a new implementation of data hash - the one we use for batch digest. The fabric implementation has a bug we don't to carry to this version.

Reply:
Issue created: https://github.com/hyperledger/fabric-x-orderer/issues/367